src/main.py

```diff
@@ -26,6 +28,41 @@
 request = Request()
 
 
+@bot.event
+async def on_member_join(member: member_type) -> None:
+    LOGGER.warning("Member %s has just joined the server", member.name)
+
+    # Get welcome channel from server
+    channel: GuildChannel = bot.get_channel(int(os.getenv("WELCOME_CHANNEL_ID")))
+
+    # Identify the path to the function and his name
+    function_name = "welcome"
+    function_path = f"{FUNCTION_BASE_RUL}{function_name}"
+    google_auth_token = google.oauth2.id_token.fetch_id_token(request, function_path)
+
+    # Call the function "welcome.go" by http request to the cloud function server and get the response
+    requests.post(
```

Comment:
`welcome` will be a pubsub function, not a http one
We need to use the `publish_message`, check `src/functions/channelMessage/cmd/publish.py` for an example

```py
import json

from google.cloud.pubsub_v1 import PublisherClient
from google.cloud.pubsub_v1.publisher.futures import Future

PROJECT_ID = "archy-f06ed"
TOPIC_ID = "TOPIC_ID"

publisher = PublisherClient()
topic_path = publisher.topic_path(PROJECT_ID, TOPIC_ID)

data = {
    "channel_id": "<INSERT_CHANNEL_ID>",
    "message": "Channel message!",
}

# Data must be a bytestring
user_encode_data = json.dumps(data, indent=2).encode("utf-8")

# When you publish a message, the client returns a future.
future: Future = publisher.publish(topic_path, user_encode_data)

print(f"Message id: {future.result()}")
print(f"Published message to {topic_path}.")
```

Reply:
I use this template inside src/main.py for on_member_join(member: member_type) function for the main node ?

Reply:
I am not certain if it's like this you mean for publilsh_message and it's like this in src/main.py for on_member_join(member: member_type)

Reply:
ok